monitoring/rid_qualifier/flight_data_generator.py

```diff
@@ -0,0 +1,481 @@
+        now = arrow.now()
+        now_isoformat = now.isoformat()
+        for i in range(num_flights):
+            flight_positions_len = len(self.grid_cells_flight_tracks[i].track)
+            
+            # in a circular flight pattern increment direction
+            angle_increment = (360 / flight_positions_len)
+            
+            # the resolution of track is 1 degree minimum
+            angle_increment = 1.0 if angle_increment == 0.0 else angle_increment
+
+            if i not in flight_track_details:
+                flight_track_details[i] = {}
+            flight_track_details[i]['track_length'] = flight_positions_len
+            
+            flight_current_index[i] = 0
+            all_flight_telemetry[i]= {}
+            all_flight_telemetry[i]['states'] = []
+
+        for j in range(duration):
+            if j == 0:
+                timestamp = now.shift(seconds=time_increment_seconds)
+            else:
+                timestamp = timestamp.shift(seconds=time_increment_seconds)
+            
+                
+            timestamp_isoformat = timestamp.isoformat()
+            
+            for k in range(num_flights): 
+                list_end = flight_track_details[k]['track_length'] - \
+                    flight_current_index[k]
+
+                if list_end != 1:
+                    flight_point = self.grid_cells_flight_tracks[k].track[flight_current_index[k]]
+                    aircraft_position = AircraftPosition(lat=flight_point.lat,
+                                                         lng=flight_point.lng,
+                                                         alt=flight_point.alt,
+                                                         accuracy_h="HAUnkown",
+                                                         accuracy_v="VAUnknown",
+                                                         extrapolated=False,
+                                                         )
+                    aircraft_height = AircraftHeight(distance=self.altitude_agl, reference="TakeoffLocation")
+
+                    rid_aircraft_state = AircraftState(
+                        timestamp=timestamp_isoformat,
+                        operational_status="Airborne",
+                        position=aircraft_position,
+                        height=aircraft_height,
+                        track=flight_point.bearing,
+                        speed=flight_point.speed,
+                        speed_accuracy="SA3mps",
+                        vertical_speed=0.0)
+
+                    all_flight_telemetry[k]['states'].append(rid_aircraft_state)
+                    
+                    flight_current_index[k] += 1
+                else:
+                    flight_current_index[k] = 0
+
+
+        telemetery_data_list = []
+        for m in range(num_flights):
+            
+            rid_aircraft_flight = RIDFlight(id=m, aircraft_type="Helicopter", states=all_flight_telemetry[m]['states'])
```

Comment:
The `id` field is a `str`, so it should be populated with `id=str(m)` rather than `id=m` (Numbers and strings are serialized differently in JSON)

Reply:
Yes, thanks for this, I have fixed it now.